src/node_errors.cc

```diff
@@ -775,8 +786,20 @@ void FatalException(Isolate* isolate,
   CHECK(!error.IsEmpty());
   HandleScope scope(isolate);
 
-  Environment* env = Environment::GetCurrent(isolate);
-  CHECK_NOT_NULL(env);  // TODO(addaleax): Handle nullptr here.
+  CHECK(isolate->InContext());
+  Local<Context> context = isolate->GetCurrentContext();
+  Environment* env = Environment::GetCurrent(context);
+  if (env == nullptr) {
+    // This could happen before Environment is assigned to the context.
+    PrintException(isolate, context, error, message);
+    // XXX(joyeecheung): this could also happen to a worker, but since
+    // we don't have access to Environment yet, there is not much
+    // we can do at this point but to exit directly.
+    isolate->TerminateExecution();
```

Comment:
I think this is pretty much all we can do at this point... @addaleax ?

Reply:
Is there even a point to calling TerminateExecution() when we're calling exit() immediately afterwards?

DisposePlatform() might block (for a while? forever?) waiting for threads to join. Maybe it'a better to abort?

Reply:
@bnoordhuis Yeah, but IIRC previously when we have errors like this in the bootstrap we usually just exit with a non-zero code. This should be quite rare and likely only happen in development, I guess it's fine to just crash here.